evalml/tests/component_tests/test_estimators.py

```diff
@@ -297,3 +299,52 @@ class MockEstimator(Estimator):
     X, y = est._manage_woodwork(X_df, y_series)
     assert isinstance(X, pd.DataFrame)
     assert isinstance(y, pd.Series)
+
+
+@pytest.mark.parametrize("estimator_class", _all_estimators())
+@pytest.mark.parametrize("use_custom_index", [True, False])
+@pytest.mark.parametrize(
+    "problem_type",
+    [ProblemTypes.BINARY, ProblemTypes.MULTICLASS, ProblemTypes.REGRESSION],
+)
+def test_estimator_fit_predict_and_predict_proba_respect_custom_indices(
```

Comment:
Why doesn't this test catch this?

```python
import pandas as pd
from evalml.demos import load_breast_cancer
from evalml.pipelines.components import LightGBMClassifier

X, y = load_breast_cancer()
lgbm = LightGBMClassifier()
X.index = range(25, 25 + X.shape[0]
lgbm.fit(X, y)

pd.testing.assert_index_equal(lgbm.predict(X).index, X.index)
```

AssertionError: Index are different

```
Index values are different (100.0 %)
[left]:  RangeIndex(start=0, stop=569, step=1)
[right]: RangeIndex(start=25, stop=594, step=1)
```

`predict_proba` does preserve the index though

Reply:
Wow, this is a really good catch! Tracking this down, it's because we have separate logic for LightGBM and XGBoost's predict methods depending on whether or not we need to encode the targets (they internally have label encoders). Since my test uses the X_y_binary fixture, it doesn't use the label encoder. In the case where we *do* use the label encoder, the label encoder's inverse_transform wipes the index so we have to set it again:

```
        X, _ = super()._manage_woodwork(X)
        X.ww.set_types(self._convert_bool_to_int(X))
        X = _rename_column_names_to_numeric(X, flatten_tuples=False)
        predictions = super().predict(X)
        if not self._label_encoder:
            return predictions
        predictions = pd.Series(
            self._label_encoder.inverse_transform(predictions.astype(np.int64)),
            index=predictions.index,
        )
        return infer_feature_types(predictions)
```


Adding another set of tests for numeric vs non-numeric targets 😅